test/prism/PrismLogicTest.scala

```diff
@@ -330,4 +330,51 @@ class PrismLogicTest extends FreeSpec with Matchers {
       }
     }
   }
-}
\ No newline at end of file
+
+  "sortInstancesByStack" - {
+    "should sort by stack, app, stage" - {
+      "when stack and stage are non empty" in {
+        val in1 = Instance("arn", "name", "state", "group", "dns","ip", DateTime.now(), "instanceName", "region", "vendor", List.empty,Map.empty, Some("stack1"), Some("stageB"), List("appB"), List.empty, Map.empty, null)
```

Comment:
It's good to try and keep the actual test cases as succinct as possible. This is fine as is, but when I find myself writing the same test boiler-plate over and over again I normally try and pull it out into an enclosing block as shared setup. Then you only need to tweak it as needed in each test case.

e.g. (disclaimer: highly contrived example)

```scala
"function I'm testing" - {
  val thing = MyThing("lengthy", "setup", "here", 1, 2, 3, etc)

  "when stack is empty" - {
    val thingWithEmptyStack = thing.copy(stack = "")

    "returns false when this condition is true" in {
      myFunction(thingWithEmptyStack, other, args) shouldEqual false
    }

    "returns true in this case" in {
      myFunction(thingWithEmptyStack, different, args) shouldEqual true
    }
  }

  "when stack has some other property" - {
    etc
  ...
}
```

This is why I prefer to use `FreeSpec`. It lets you arbitrarily create blocks that have helpful labels on them and then at the bottom you have the actual test case, which is marked with `in`. The message that the test runner gets is the full path to the actual test case e.g. `"function I'm testing when stack is empty returns false when this condition is true"` (contrived example, but hopefully you see what I mean). you can group your tests with shared setup and the test messages when they run make sense.

With this in mind if you have a look around the existing tests you'll see why they're laid out the way they are. It's about grouping tests in a way that makes them easy to understand for the reader because the use case of comprehending existing test cases is the most important one to address when writing tests.